sklearn/_config.py

```diff
@@ -2,13 +2,23 @@
 """
 import os
 from contextlib import contextmanager as contextmanager
+import threading
 
 _global_config = {
     'assume_finite': bool(os.environ.get('SKLEARN_ASSUME_FINITE', False)),
     'working_memory': int(os.environ.get('SKLEARN_WORKING_MEMORY', 1024)),
     'print_changed_only': True,
     'display': 'text',
 }
+_threadlocal = threading.local()
+
+
+def _get_threadlocal_config():
+    """Get the configuration that is local to the thread. If the configuration
+    does not exist, copy the default global configuration."""
+    if not hasattr(_threadlocal, 'global_config'):
+        _threadlocal.global_config = _global_config.copy()
+    return _threadlocal.global_config
```

Comment:
just a nitpick. Why not put that directly in get_config ?
also I think the copy is unecessary here since it there will be a copy afterwards anyway

Reply:
I thought of `_get_threadlocal_config` for returning a threadlocal **mutable** version of the config, while `get_config` returns a **immutable** copy of the config. This is a safe guard from someone doing the following:

```python
config = get_config()

# Does not change the configuration
config["assume_finite"] = True
```

which is consistent with the current behavior on `main`.

I added comments to describe this reasoning into the PR.

Reply:
But do we intend to use ``_get_threadlocal_config`` outside of ``get_config`` ? If not why don't we do

```py
def get_config():
    if not hasattr(_threadlocal, 'global_config'):
        _threadlocal.global_config = _global_config
    return  _threadlocal.global_config.copy()
```

Reply:
`_get_threadlocal_config` is used in `set_config` to update the threadlocal config.

In any case, for `set_config` to not mutable `_global_config`, `_threadlocal.global_config` needs to be initialize with a copy of `_global_config`.

Reply:
> _get_threadlocal_config is used in set_config 

Yep missed that. Sorry. Forget my comments :)